### PR TITLE
feat(textfield): add password option on textfield

### PR DESCRIPTION
**Pull request by leangseu**
## Description

Added password option to textfield. I would have gone with type attribute in input tag ( text|email|tel|password|url|number ), if it textfield does not include textarea.

## Motivation and Context

I wanted to have password field and did not want to work around with dom manipulation.

## How Has This Been Tested?

Yes, I ran `yarn test`

The coverage result:
`98.46% Statements 12471/1266694.63% Branches 1709/180694.69% Functions 589/62298.46% Lines 12471/12666`

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] My change requires a change to the documentation.
-   [x] I have updated the documentation accordingly.
-   [x] I have read the **CONTRIBUTING** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: packages/textfield/src/spectrum-config.js [hunk 1]
```diff
@@ -65,6 +65,11 @@ const config = {
                     selector: '.is-disabled',
                     name: 'disabled',
                 },
+                {
```
Comment (Westbrook) [inline packages/textfield/src/spectrum-config.js:68]: `.is-password` is not a part of the Spectrum CSS styling API so it won't need to be a part of this config.

File: packages/textfield/stories/textarea.stories.ts [hunk 2]
```diff
@@ -68,5 +68,17 @@ export const Default = (): TemplateResult => {
             disabled
             placeholder="Enter your life story"
         ></sp-textfield>
+        <sp-textfield
```
Comment (Westbrook) [inline packages/textfield/stories/textarea.stories.ts:71]: Please remove these, as there is not `multiline` version of `password`.

File: packages/textfield/stories/textfield.stories.ts [hunk 3]
```diff
@@ -48,6 +48,17 @@ export const Default = (): TemplateResult => {
             value="Not a valid input"
             disabled
         ></sp-textfield>
+        <sp-textfield
```
Comment (Westbrook) [inline packages/textfield/stories/textfield.stories.ts:51]: Please list the `type=password` examples here as their own story.

File: packages/textfield/textarea.md [hunk 4]
```diff
@@ -72,3 +72,16 @@ By default the text area has a fixed height and will scroll when text entry goes
     value="By default the text area has a fixed height and will scroll when text entry goes beyond the available space. With the use of the `grows` attribute the text area will grow to accomidate the full content of the element."
 ></sp-textfield>
 ```
+
+### Password
+
+Provide a way to securely input a password. ( Not recommend ) [See browser compatibility](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-text-security#Browser_compatibility)
```
Comment (Westbrook) [inline packages/textfield/textarea.md:78]: I don't see any changes that would apply the API outlined in this URL to this change, was this reference intended?

**Review by Westbrook — COMMENTED** [review]
@leangseu, thanks for submitting this PR! I've listed some changes that I think would be needed before we could accept this change in its current state.

However, I'm not sure we've clarified a path forward for working with these sorts of additions to the Textfield API (`text|email|tel|password|url|number`) enough for this change to be accepted in that state. Particularly, the idea that you decided not to add the entirety of that list due to the component's current structure points to some large structural changes that should be discussed before adding support for this.

It would be greatly appreciated if you were to open an issue discussing this addition so we could work out the appropriate API to support this sort of feature, and the possibility of others in a similar fashion, going forward.

File: packages/textfield/textarea.md (hunk 4 above)
Comment (leangseu) [reply packages/textfield/textarea.md:78]: This was for the is-password css. I was planning to have multiline password option. However, I am happy with having type="password" instead.

File: PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,45 +1,54 @@
...
-<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
-
-- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-- [ ] My code follows the code style of this project.
-- [ ] My change requires a change to the documentation.
-- [ ] I have updated the documentation accordingly.
-- [ ] I have read the **CONTRIBUTING** document.
-- [ ] I have added tests to cover my changes.
-- [ ] All new and existing tests passed.
+<!--- Provide a general summary of your changes in the Title above -->
```
[the remark this answers is not in the document]
Comment (leangseu) [reply PULL_REQUEST_TEMPLATE.md:1]: Sorry, this is my first time contributing. I thought I had to fill that out.

**Pull request closed without merging**